packages/solidity-contracts/test/behaviors/erc20GatewayV4.behavior.test.ts

```diff
@@ -0,0 +1,947 @@
+import type { HardhatEthersSigner } from '@nomicfoundation/hardhat-ethers/signers';
+import { setBalance } from '@nomicfoundation/hardhat-network-helpers';
+import { expect } from 'chai';
+import {
+  MaxUint256,
+  ZeroHash,
+  parseEther,
+  parseUnits,
+  solidityPacked,
+  zeroPadValue,
+} from 'ethers';
+import type { BytesLike } from 'ethers';
+import hre from 'hardhat';
+
+import { CONTRACT_MESSAGE_PREDICATE } from '../../protocol/constants';
+import { randomBytes32 } from '../../protocol/utils';
+import { CustomToken__factory } from '../../typechain';
+import type {
+  MockFuelMessagePortal,
+  FuelERC20GatewayV4,
+  Token,
+  CustomToken,
+} from '../../typechain';
+import { impersonateAccount } from '../utils/impersonateAccount';
+
+type Env = {
+  fuelMessagePortal: MockFuelMessagePortal;
+  assetIssuerId: BytesLike;
+  erc20Gateway: FuelERC20GatewayV4;
+  token: Token;
+  signers: HardhatEthersSigner[];
+  deployer: HardhatEthersSigner;
+};
+
+enum MessageType {
+  DEPOSIT_TO_ADDR = 0,
+  DEPOSIT_TO_CONTRACT = 1,
+  DEPOSIT_WITH_DATA = 2,
+  METADATA = 3,
+}
+
+const MessagePayloadSolidityTypes = [
+  'uint256', // assetIssuerId
+  'uint256', // message type
+  'uint256', // tokenAddress
+  'uint256', // tokenId
+  'uint256', // depositor EVM address, padded
+  'uint256', // recipient FuelVM address
+  'uint256', // l2 amount to be minted
+];
+
+export function behavesLikeErc20GatewayV4(fixture: () => Promise<Env>) {
+  describe('Behaves like FuelERC20GatewayV4', () => {
+    let env: Env;
+
+    beforeEach('reset fixture', async () => {
+      env = await fixture();
+    });
+
+    describe('setGlobalDepositLimit()', () => {
+      it('reverts when called by an unauthorized address', async () => {
+        const {
+          erc20Gateway,
+          token,
+          signers: [, mallory],
+        } = env;
+
+        const tx = erc20Gateway
+          .connect(mallory)
+          .setGlobalDepositLimit(token, MaxUint256);
+
+        const expectedErrorMsg =
+          `AccessControl: account ${(
+            await mallory.getAddress()
+          ).toLowerCase()} ` +
+          'is missing role 0x0000000000000000000000000000000000000000000000000000000000000000';
+        await expect(tx).to.be.revertedWith(expectedErrorMsg);
+      });
+
+      it('changes the deposit limit', async () => {
+        const {
+          erc20Gateway,
+          token,
+          signers: [deployer],
+        } = env;
+
+        await erc20Gateway
+          .connect(deployer)
+          .setGlobalDepositLimit(token, MaxUint256);
+
+        expect(await erc20Gateway.depositLimits(token)).to.be.eq(MaxUint256);
+      });
+    });
+
+    // describe.skip('finalizeWithdrawal', () => {
```

Comment:
What's required for this to be re-introduced?

Reply:
No, this methods are gone. Will remove the commented out code